neuralpredictors/layers/cores/conv2d.py

```diff
@@ -542,3 +541,165 @@ def initialize(self):
+            skip:           Adds a skip connection
+            final_nonlinearity: Boolean, if true, appends an ELU layer after the last BatchNorm (if BN=True)
+            elu_xshift, elu_yshift: final_nonlinearity(x) = Elu(x - elu_xshift) + elu_yshift
+            bias:           Adds a bias layer.
+            momentum:       BN momentum
+            pad_input:      Boolean, if True, applies zero padding to all convolutions
+            batch_norm:     Boolean, if True appends a BN layer after each convolutional layer
+            hidden_dilation:    If set to > 1, will apply dilated convs for all hidden layers
+            laplace_padding: Padding size for the laplace convolution. If padding = None, it defaults to half of
+                the kernel size (recommended). Setting Padding to 0 is not recommended and leads to artefacts,
+                zero is the default however to recreate backwards compatibility.
+            input_regularizer:  String that must match one of the regularizers in ..regularizers
+            stack:        Int or iterable. Selects which layers of the core should be stacked for the readout.
+                            default value will stack all layers on top of each other.
+                            Implemented as layers_to_stack = layers[stack:]. thus:
+                                stack = -1 will only select the last layer as the readout layer.
+                                stack of -2 will read out from the last two layers.
+                                And stack of 1 will read out from layer 1 (0 indexed) until the last layer.
+            se_reduction:   Int, Reduction of channels for global pooling of the Squeeze and Excitation Block.
+            n_se_blocks:    Int, number of squeeze and excitation blocks. Inserted from the last layer
+                              Examples: layers=4, n_se_blocks=2:
+                                => layer0 -> layer1->layer2->SEblock->layer3->SEblock
+            depth_separable: Boolean, if True, uses depth-separable convolutions in all layers after the first one.
+            attention_conv: Boolean, if True, uses self-attention instead of convolution for all layers after the first one.
+            linear:         Boolean, if True, removes all nonlinearities
+        """
+
+        super().__init__()
+
+        assert not bias or not batch_norm, "bias and batch_norm should not both be true"
+        assert not depth_separable or not attention_conv, "depth_separable and attention_conv should not both be true"
+
+        regularizer_config = (
+            dict(padding=laplace_padding, kernel=input_kern)
+            if input_regularizer == "GaussianLaplaceL2"
+            else dict(padding=laplace_padding)
+        )
+        self._input_weights_regularizer = regularizers.__dict__[input_regularizer](**regularizer_config)
+
+        self.layers = layers
```

Comment:
I'd call it `self.num_layers` and `num_layers` or something. To make clear that this is an integer.